specs/zkevm-prover.md

```diff
@@ -9,117 +9,43 @@
-      nonce: 0,
-      gasLimit: 0,
-      gasPrice: 0,
-      to: constants.AddressZero,
-      value: 0,
-      data: '0x',
-      chainId,
-    });
-    return keccak256(rlp);
-  }
-
-  async function computePublicInput(block: RPCBlock, chainId: number): Promise<[string, string]> {
-    const maxTxs = 25;
-
-    const buf = Buffer.concat([
-      hex.toFixedBuffer(prevStateRoot, 32),
-      hex.toFixedBuffer(block.stateRoot, 32),
-      hex.toFixedBuffer(block.withdrawalsRoot ?? 0, 32),
-      hex.toFixedBuffer(block.hash, 32),
-      hex.toFixedBuffer(block.parentHash, 32),
-      hex.toFixedBuffer(block.number, 8),
-      hex.toFixedBuffer(block.timestamp, 8),
-      hex.toFixedBuffer(block.baseFeePerGas ?? 0, 32),
-      hex.toFixedBuffer(block.gasLimit, 8),
-      hex.toFixedBuffer(block.transactions.length, 2),
-      Buffer.concat(
-        block.transactions.map((txHash: string) => {
-            return toFixedBuffer(txHash, 32);
-        }),
-      ),
-      Buffer.concat(
-        Array(maxTxs - block.transactions.length).fill(
-          toFixedBuffer(await getDummyTxHash(chainId), 32),
-        ),
-      ),
-    ]);
-    const h = hex.toFixedBuffer(keccak256(buf), 32);
-    return [
-      '0x' + h.subarray(0, 16).toString('hex'),
-      '0x' + h.subarray(16, 32).toString('hex'),
-    ];
-  }
-  ```
-
-- Witness: TODO
-
-> These relations are changed soon after [super circuit] integration.
-
-[super circuit]: https://github.com/kroma-network/zkevm-specs/blob/dev/specs/super_circuit.png
-
-## the ZK Verifier Contract
-
-The ZK Verifier contract implements `verify` function like followings:
+The Prover is responsible for generating proofs using the [Halo2 proving scheme](https://zcash.github.io/halo2/)
+for target blocks. These proofs provide claims regarding the legitimacy of block data and state transitions.
+And the prover is main feature of the challenge process as a component of challenger.
+
+[ZK fault proof][g-zk-fault-proof] that states that a [state][g-state] transition from `S` to
+`S'` is valid. It sounds like there are no big differences from validity proofs. That's true. But the point is
+this is used to prove the state transition `S` to `S''` is wrong by showing a valid state transition `S` to `S'`.
+
+
+## ZKEVM Proofs
+
+In the context of the EVM, which is regarded as a Turing machine, the generation of proofs requires demonstrating the
+execution of Opcode operations in arbitrary order. These proofs serve as evidence of computation for the EVM.
+Additionally, validating the block data's integrity involves proving sub-operations such as Merkle Patricia Trie
+and ECDSA verification.
+To support the proof generation process for target blocks, [zkevm-circuits] offer circuits capable of proving the
+required sub-operations. These circuits provide the means to prove the integrity and validity of the block data.
+By combining these circuits, the Prover completes the overall validity proof for the target block.  
```

Comment:
It would be better to add an empty line here to improve readability.

Reply:
applied